src/services/CommandImportService.ts

```diff
@@ -0,0 +1,34 @@
+import { APIApplicationCommand, REST, Routes } from "discord.js";
+import { ImportedClass } from "../helpers/typeHelpers";
+import NeedleCommand from "../models/NeedleCommand";
+import DynamicImportService from "./DynamicImportService";
+
+export default class CommandImportService extends DynamicImportService<typeof NeedleCommand> {
+	private idCache: { id: string; name: string }[] = [];
+
+	public async load(skipCache = false): Promise<ImportedClass<typeof NeedleCommand>[]> {
+		if (this.idCache.length === 0) {
+			this.idCache = await this.fetchGlobalApplicationCommands();
+		}
+		return super.load(skipCache);
+	}
+
+	public getId(commandName: string): string {
+		const id = this.idCache.find(cmd => cmd.name === commandName)?.id;
```

Comment:
doesn't find return `-1` and not null?

Reply:
![image](https://user-images.githubusercontent.com/55610086/187321041-522f2397-8b08-4b94-a07b-ec226631f060.png)
wrong method